columnflow/tasks/framework/inference.py

```diff
@@ -106,153 +107,184 @@ def get_data_datasets(cls, config_inst: od.Config, cat_obj: DotDict) -> list[str
+                    mc_datasets = self.get_mc_datasets(config_inst, proc_obj)
+                    for dataset_name in mc_datasets:
+                        if dataset_name not in data["mc_datasets"]:
+                            data["mc_datasets"][dataset_name] = {
+                                "proc_name": proc_obj.name,
+                                "shift_sources": set(),
+                            }
+                        elif data["mc_datasets"][dataset_name]["proc_name"] != proc_obj.name:
+                            raise ValueError(
+                                f"dataset '{dataset_name}' was already assigned to datacard process "
+                                f"'{data['mc_datasets'][dataset_name]['proc_name']}' and cannot be re-assigned to "
+                                f"'{proc_obj.name}' in config '{config_inst.name}'",
+                            )
 
-    def _requires_cat_obj(self, cat_obj: DotDict, merge_variables: bool = False, **req_kwargs):
-        """
-        Helper to create the requirements for a single category object.
+                    # shift sources
+                    for param_obj in proc_obj.parameters:
+                        if config_inst.name not in param_obj.config_data:
+                            continue
+                        # only add if a shift is required for this parameter
+                        if param_obj.type.is_shape or any(trafo.from_shape for trafo in param_obj.transformations):
+                            shift_source = param_obj.config_data[config_inst.name].shift_source
+                            for mc_dataset in mc_datasets:
+                                data["mc_datasets"][mc_dataset]["shift_sources"].add(shift_source)
+
+                # for consistency checks later
+                variables.add(cat_obj.config_data[config_inst.name].variable)
+                categories.add(cat_obj.config_data[config_inst.name].category)
+
+            # consistency checks: the config-based variable and category names must be identical
+            if len(variables) != 1:
+                raise ValueError(
+                    f"found diverging variables to be used in datacard category '{cat_obj.name}' across configs "
+                    f"{', '.join(c.name for c in config_insts)}: {variables}",
+                )
+            if len(categories) != 1:
+                raise ValueError(
+                    f"found diverging categories to be used in datacard category '{cat_obj.name}' across configs "
+                    f"{', '.join(c.name for c in config_insts)}: {categories}",
+                )
 
-        :param cat_obj: category object from an InferenceModel
-        :param merge_variables: whether to merge the variables from all requested category objects
-        :return: requirements for the category object
-        """
+        return config_data
+
+    def create_branch_map(self):
+        # dummy branch map
+        return {0: None}
```

Comment:
As discussed offline, just to keep it documented:
It might be nice (in a separate PR) to add the inference model categories to the branch data (branch map with length 1) & loop over the categories from the branch data. That way, one could overwrite the branch map from analysis side to reimplement the parallelization